src/main/scala/loamstream/model/execute/ExecutionCell.scala

```diff
@@ -0,0 +1,52 @@
+package loamstream.model.execute
+
+import loamstream.model.jobs.LJob
+import loamstream.model.jobs.JobStatus
+import loamstream.model.jobs.JobResult
+
+/**
+ * @author clint
+ * Jan 24, 2020
+ * 
+ * A snapshot of an LJob's state at a particular moment during execution, from the perspective of RxExecuter and
+ * ExecutionState.  Tracks the jobs status and the number of times it has ran. 
+ */
+final case class ExecutionCell(status: JobStatus = JobStatus.NotStarted, runCount: Int = 0) {
+  
+  def startRunning: ExecutionCell = copy(status = JobStatus.Running, runCount = runCount + 1)
+  
+  def markAsRunnable: ExecutionCell = copy(status = JobStatus.NotStarted)
+  
+  def finishWith(newStatus: JobStatus): ExecutionCell = {
+    require(newStatus.isFinished, s"Expected finished job status, but got $status")
+    
+    copy(status = newStatus)
+  }
+  
+  def markAs(newStatus: JobStatus): ExecutionCell = {
```

Comment:
I like this, especially since it can be used (like it is) to confirm valid changes from one status to another. But I don't see it being used (internally) for things like `markAsRunnable` or `startRunning`, where it seems it would be valuable. Can I `startRunning` before it's marked as runnable?

Reply:
As it stands, yes.  Should you be able to?  Arguably not.  `ExecutionCell` evolved to sort-of encode a FSM, at least by enforcing invariants in a side-effecty way.  I'll see if it's feasible to assert that a job is marked as not started before running it.  

I "encoded" the FSM this way, such as it is, since the set of all state transitions gets very complex very quickly, when considering multiple execution environments: Local, Uger, LSF, Google, Amazon.  Still, it might be possible to have a more-reified FSM.  When I was writing this, I was doing major-enough surgery that I was anxious about getting it done in a timely manner, which made me skip over any nicer FSM encoding.  (Not to toot my own horn, but I re-did the core execution engine of LS in less than a week.)